src/aosm/azext_aosm/_configuration.py

```diff
@@ -348,85 +417,83 @@ def validate(self):
 
         :raises ValidationError: If source registry ID doesn't match the regex
         """
+        assert isinstance(self.images, CNFImageConfig)
+        super().validate()
 
-        source_reg_set = self.images.source_registry != ""
-        source_local_set = self.images.source_local_docker_image != ""
-        source_reg_namespace_set = self.images.source_registry_namespace != ""
+        self.images.validate()
 
-        if source_reg_namespace_set and not source_reg_set:
-            raise ValidationError(
-                "Config validation error. The image source registry namespace should "
-                "only be configured if a source registry is configured."
-            )
-        # If these are the same, either neither is set or both are, both of which are errors
-        if source_reg_set == source_local_set:
-            raise ValidationError(
-                "Config validation error. Images config must have either a local docker image"
-                " or a source registry, but not both."
-            )
-
-
-NFD_NAME = "The name of the existing Network Function Definition Group to deploy using this NSD"
-NFD_VERSION = (
-    "The version of the existing Network Function Definition to base this NSD on.  "
-    "This NSD will be able to deploy any NFDV with deployment parameters compatible "
-    "with this version."
-)
-NFD_LOCATION = "The region that the NFDV is published to."
-PUBLISHER_RESOURCE_GROUP = "The resource group that the publisher is hosted in."
-PUBLISHER_NAME = "The name of the publisher that this NFDV is published under."
-PUBLISHER_SCOPE = (
-    "The scope that the publisher is published under. Only 'private' is supported."
-)
-NFD_TYPE = "Type of Network Function. Valid values are 'cnf' or 'vnf'"
-MULTIPLE_INSTANCES = (
-    "Set to true or false.  Whether the NSD should allow arbitrary numbers of this "
-    "type of NF.  If set to false only a single instance will be allowed.  Only "
-    "supported on VNFs, must be set to false on CNFs."
-)
+        for helm_package in self.helm_packages:
+            assert isinstance(helm_package, HelmPackageConfig)
+            helm_package.validate()
 
 
 @dataclass
 class NFDRETConfiguration:  # pylint: disable=too-many-instance-attributes
     """The configuration required for an NFDV that you want to include in an NSDV."""
 
-    publisher: str = PUBLISHER_NAME
-    publisher_resource_group: str = PUBLISHER_RESOURCE_GROUP
-    name: str = NFD_NAME
-    version: str = NFD_VERSION
-    publisher_offering_location: str = NFD_LOCATION
-    publisher_scope: str = PUBLISHER_SCOPE
-    type: str = NFD_TYPE
-    multiple_instances: Union[str, bool] = MULTIPLE_INSTANCES
+    publisher: str = ""
+    publisher_resource_group: str = ""
+    name: str = ""
+    version: str = ""
+    publisher_offering_location: str = ""
+    publisher_scope: str = ""
+    type: str = ""
+    multiple_instances: Union[str, bool] = False
```

Comment:
QoL improvement - since this remains a Union since we're using the same class for import/export and we want a bool on import and a string on export, it would be nice if we could handle strings that match "true" or "false" to save a cycle of validation error and retry for the user when they inevitably don't delete the quotes from the exported version

Reply:
Great idea, thanks!  Done.